plus/src/main/java/org/apache/calcite/chinook/ChinookAvaticaServer.java

```diff
@@ -62,16 +62,18 @@ public static class CalciteChinookMetaFactory implements Meta.Factory {
     private static final CalciteConnectionProvider CONNECTION_PROVIDER =
         new CalciteConnectionProvider();
 
-    private static JdbcMeta instance = null;
+    private static volatile JdbcMeta instance = null;
 
     private static JdbcMeta getInstance() {
       if (instance == null) {
-        try {
-          instance =
-              new JdbcMeta(CalciteConnectionProvider.DRIVER_URL,
-                  CONNECTION_PROVIDER.provideConnectionInfo());
-        } catch (SQLException | IOException e) {
-          throw new RuntimeException(e);
+        synchronized (JdbcMeta.class) {
```

Comment:
How about just synchronizing on `CalciteChinookMetaFactory.class`? (Same to `RawChinookMetaFactory`)

Reply:
@chucheng92 hi, benchao. thanks for reviewing. updated.